test/test_rng_fail.c

```diff
@@ -0,0 +1,185 @@
+                  "(expected %d) when randombytes %d/%d was "          \
+                  "instrumented to fail\n",                            \
+                  test_name, rc, MLD_ERR_RNG_FAIL, i + 1,              \
+                  num_randombytes_calls);                              \
+        }                                                              \
+        return 1;                                                      \
+      }                                                                \
+    }                                                                  \
+    printf(                                                            \
+        "RNG failure test for %s PASSED.\n"                            \
+        "  Tested %d randombytes invocation point(s)\n",               \
+        test_name, num_randombytes_calls);                             \
+  } while (0)
+
+static int test_keypair_rng_failure(void)
+{
+  uint8_t pk[MLDSA_PUBLICKEYBYTES(MLD_CONFIG_API_PARAMETER_SET)];
+  uint8_t sk[MLDSA_SECRETKEYBYTES(MLD_CONFIG_API_PARAMETER_SET)];
+
+  TEST_RNG_FAILURE("crypto_sign_keypair", mld_keypair(pk, sk));
+  return 0;
+}
+
+static int test_signature_rng_failure(void)
+{
+  uint8_t pk[MLDSA_PUBLICKEYBYTES(MLD_CONFIG_API_PARAMETER_SET)];
+  uint8_t sk[MLDSA_SECRETKEYBYTES(MLD_CONFIG_API_PARAMETER_SET)];
+  uint8_t sig[MLDSA_BYTES(MLD_CONFIG_API_PARAMETER_SET)];
+  size_t siglen;
+  uint8_t msg[32] = {0};
+  uint8_t ctx[10] = "test";
+
+  /* Generate valid keypair first */
+  reset_all();
+  if (mld_keypair(pk, sk) != 0)
+  {
+    fprintf(stderr,
+            "ERROR: crypto_sign_keypair failed in signature test setup\n");
+    return 1;
+  }
+
+  TEST_RNG_FAILURE(
+      "crypto_sign_signature",
+      mld_signature(sig, &siglen, msg, sizeof(msg), ctx, sizeof(ctx), sk));
+  return 0;
+}
+
+static int test_signature_extmu_rng_failure(void)
+{
+  uint8_t pk[MLDSA_PUBLICKEYBYTES(MLD_CONFIG_API_PARAMETER_SET)];
+  uint8_t sk[MLDSA_SECRETKEYBYTES(MLD_CONFIG_API_PARAMETER_SET)];
+  uint8_t sig[MLDSA_BYTES(MLD_CONFIG_API_PARAMETER_SET)];
+  size_t siglen;
+  uint8_t mu[MLDSA_CRHBYTES];
+
+  /* Generate valid keypair first */
+  reset_all();
+  if (mld_keypair(pk, sk) != 0)
+  {
+    fprintf(
+        stderr,
+        "ERROR: crypto_sign_keypair failed in signature_extmu test setup\n");
+    return 1;
+  }
+
+  /* Fill mu with test data */
+  randombytes(mu, sizeof(mu));
+
+  TEST_RNG_FAILURE("crypto_sign_signature_extmu",
+                   mld_signature_extmu(sig, &siglen, mu, sk));
+  return 0;
+}
+
+int main(void)
+{
+  printf("Testing ML-DSA parameter set %d\n", MLD_CONFIG_API_PARAMETER_SET);
+  printf("========================================\n");
+
+  if (test_keypair_rng_failure() != 0)
+  {
+    return 1;
+  }
+
+  if (test_signature_rng_failure() != 0)
+  {
+    return 1;
+  }
+
+  if (test_signature_extmu_rng_failure() != 0)
+  {
+    return 1;
+  }
```

Comment:
I'd be inclined to just cover the same set of functions as in `test_alloc.c`, regardless of whether (currently) they call RNG or not. For example, `crypto_sign` also indirectly uses RNG, and the present test would not catch a bad error propagation in the call path from `crypto_sign -> crypto_sign_signature`.